src/TrajectoryPlanner/src/TrajectoryGenerator.cpp

```diff
@@ -496,6 +542,22 @@ bool TrajectoryGenerator::isTrajectoryAsked()
     return m_generatorState == GeneratorState::Called;
 }
 
+bool TrajectoryGenerator::generateTrajectoriesFromFootprints(std::shared_ptr<FootPrint> left, std::shared_ptr<FootPrint> right,
+                                                             const double &initTime)
+{
+    DCMInitialState initialState;
+    initialState.initialPosition = m_DCMBoundaryConditionAtMergePointPosition;
+    initialState.initialVelocity = m_DCMBoundaryConditionAtMergePointVelocity;
+
+    if (!m_dcmGeneratorStepAdj->setDCMInitialState(initialState))
+    {
+        yError() << "[TrajectoryGenerator_Thread] Failed to set the initial state.";
+        return  false;
+    }
+
+    return m_trajectoryGeneratorStepAdj.generateFromFootPrints(left, right, initTime, m_dT);
```

Comment:
Notice that this call is blocking. How much time does it take to compute?

Reply:
Actually I'm not sure that this part of the code will be run! Because I tried to see the execution time but it does not publish anything!

Reply:
I tried to calculate the  execution time of this part inside the walking module(since it does not publish something on the terminal when I use it inside the` TrajectoryGenerator.cpp `), that also include the execution time of the other parts of the code that is less than 500 micro second.

Reply:
Ok great! I guess that for the time being it's not necessary to take further action. It this is going to be a problem in the future, we can address it accordingly. Maybe can you just drop a line in the documentation? Like "Note, this call is blocking."

Reply:
Done!